### PR TITLE
[Merged by Bors] - chore: unify monotonicity of rank lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
The names were all over the place. Rename:
* `rank_le_of_submodule` -> `Submodule.rank_mono`
* `Submodule.finrank_le_finrank_of_le` -> `Submodule.finrank_mono` (existing, up to unfolding `Monotone`)
* `rank_submodule_le` -> `Submodule.rank_le`

From LeanCamCombi

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dimension/Basic.lean [hunk 1]
```diff
@@ -333,7 +334,7 @@ theorem rank_range_of_surjective (f : M →ₗ[R] M') (h : Surjective f) :
 
 theorem rank_submodule_le (s : Submodule R M) : Module.rank R s ≤ Module.rank R M := by
   rw [← rank_top R M]
-  exact rank_le_of_submodule _ _ le_top
```
Comment (grunweg) [inline Mathlib/LinearAlgebra/Dimension/Basic.lean:336]: Should this lemma be called `Submodule.rank_le` instead? You're renaming the lemma above to `Submodule.rank_mono`...

Actually: the analogous lemma for finrank is called `Submodule.finrank_le` - please rename.

File: Mathlib/LinearAlgebra/FiniteDimensional/Defs.lean [hunk 2]
```diff
@@ -694,21 +693,6 @@ noncomputable def fieldOfFiniteDimensional (F K : Type*) [Field F] [h : CommRing
...
 end
-
-namespace Submodule
-
-section DivisionRing
-
-variable [DivisionRing K] [AddCommGroup V] [Module K V] {V₂ : Type v'} [AddCommGroup V₂]
-  [Module K V₂]
-
-theorem finrank_mono [FiniteDimensional K V] : Monotone fun s : Submodule K V => finrank K s :=
```
Comment (grunweg) [inline Mathlib/LinearAlgebra/FiniteDimensional/Defs.lean:705]: This is subsumed and generalised by the new lemma called `Submodule.finrank_mono`. Can you mention this in the commit message, please? (Or restore this lemma, appropriately named?)

**Review by grunweg — COMMENTED** [review]
Thanks! I have two small comments.

File: Mathlib/LinearAlgebra/FiniteDimensional/Defs.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/LinearAlgebra/FiniteDimensional/Defs.lean:705]: Mentioned in the PR description

File: Mathlib/LinearAlgebra/Dimension/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/LinearAlgebra/Dimension/Basic.lean:336]: Renamed

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**